modules/dataquery/ajax/saveQuery.php

```diff
@@ -20,8 +20,29 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize(__DIR__ . "/../../../project/config.xml");
-$user         = User::singleton();
+
+$user = User::singleton();
+$cdb  = CouchDB::singleton();
+$qid  = $user->getUserName() . "_" . $_REQUEST['QueryName'];
+
+if ($_REQUEST['SharedQuery'] === "true") {
+    $qid = "global:" . $qid;
+}
+
+if ($_REQUEST['OverwriteQuery'] === "false") {
+    $results = $cdb->getDoc(
+        $qid
```

Comment:
Alright I'll change the status of my review; I don't want to block this without a proven exploit.

I'll make a note to test the DQT at a later time to test my suspicions.